### PR TITLE
chore(codepipeline-actions,rds): migrate tests to jest

### PR DESCRIPTION
**Pull request by nija-at**
Move codepipeline-actions from nodeunit to jest. Most tests
now use `nodeunit-shim` but some tests have been migrated
entirely to pure jest.

Move `cluster.test.ts` and `instance.test.ts` in the rds module
from `nodeunit-shim` to native jest.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-codepipeline-actions/test/lambda/lambda-invoke-action.test.ts [hunk 1]
```diff
@@ -172,18 +174,18 @@ export = {
...
-      }));
+      });
+
 
-      test.done();
-    },
+    });
 
-    "assigns the Action's Role with read-write permissions to the Bucket if it has both inputs and outputs"(test: Test) {
+    testFutureBehavior("assigns the Action's Role with read-write permissions to the Bucket if it has both inputs and outputs", s3GrantWriteCtx, App, (app) => {
```
Comment (njlynch) [inline packages/@aws-cdk/aws-codepipeline-actions/test/lambda/lambda-invoke-action.test.ts:182]: General comment. What's your strategy here with testFutureBehavior on v1? Are we proactively dropping tests for "legacy" behavior already, or supposed to have tests for both behaviors, or some other mix? Feels like by converting this test to `testFutureBehavior`, we are no longer testing the "legacy" (read: default) behavior.

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts [hunk 2]
```diff
@@ -1831,15 +1828,16 @@ nodeunitShim({
...
-    }));
+    });
 
-    test.done();
-  },
+
+  });
 });
 
+<<<<<<< HEAD
```
Comment (njlynch) [inline packages/@aws-cdk/aws-rds/test/cluster.test.ts:1840]: ❗  Whoops!

**Review by njlynch — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-codepipeline-actions/test/lambda/lambda-invoke-action.test.ts (hunk 1 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-codepipeline-actions/test/lambda/lambda-invoke-action.test.ts:182]: I don't have a general strategy. It's more applying discretion case-by-case.

In this specific case, the feature flag is changing policy action from `s3:PutObject*` to `s3:PutObject`.
This is quite well tested in the s3 module, so in consuming modules (such as the ones in this PR), I'm only testing the future behaviour. Feels redundant to test the legacy behaviour.

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts (hunk 2 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-rds/test/cluster.test.ts]: whoops indeed.

**Review by njlynch — APPROVED** [review]

**Pull request merged**